ip_country/src/country_finder.rs

```diff
@@ -225,77 +191,84 @@ mod tests {
 
+        assert_eq!(result.free_world, true);
+        assert_eq!(result.iso3166, "US".to_string());
         assert_eq!(result.name, "United States".to_string());
-        let duration = timeend.duration_since(timestart).unwrap().as_secs();
-        println!("duration: {}", duration);
-        assert!(duration <= 5);
+        let duration = time_end.duration_since(time_start).unwrap();
+        assert!(
+            duration.as_secs() < 1,
+            "Duration of the search was too long: {} ms",
+            duration.as_millis()
+        );
     }
 
     #[test]
     fn deserialize_country_blocks_ipv4_and_ipv6_adn_fill_vecs() {
-        let mut result_ipv4: Vec<CountryBlock> = vec![];
-        let mut result_ipv6: Vec<CountryBlock> = vec![];
-        let timestart = SystemTime::now();
-        let mut deserializer_ipv4 =
-            CountryBlockDeserializerIpv4::new(crate::dbip_country::ipv4_country_data());
-        let mut deserializer_ipv6 =
-            CountryBlockDeserializerIpv6::new(crate::dbip_country::ipv6_country_data());
-        let timeend = SystemTime::now();
+        let time_start = SystemTime::now();
+        let deserializer_ipv4 = CountryBlockDeserializer::<Ipv4Addr, u8, 4>::new(
+            crate::dbip_country::ipv4_country_data(),
+        );
+        let deserializer_ipv6 = CountryBlockDeserializer::<Ipv6Addr, u16, 8>::new(
+            crate::dbip_country::ipv6_country_data(),
+        );
+        let time_end = SystemTime::now();
 
-        let timestart_fill = SystemTime::now();
-        loop {
-            match deserializer_ipv4.next() {
-                None => break, // this line isn't really testable, since the deserializer will produce CountryBlocks for every possible address
-                Some(country_block) => {
-                    result_ipv4.push(country_block);
-                }
-            }
-        }
-        loop {
-            match deserializer_ipv6.next() {
-                None => break, // this line isn't really testable, since the deserializer will produce CountryBlocks for every possible address
-                Some(country_block) => {
-                    result_ipv6.push(country_block);
-                }
-            }
-        }
-        let timeend_fill = SystemTime::now();
+        let time_start_fill = SystemTime::now();
+        let _ = deserializer_ipv4.collect_vec();
+        let _ = deserializer_ipv6.collect_vec();
+        let time_end_fill = SystemTime::now();
 
-        let duration = timeend.duration_since(timestart).unwrap().as_secs();
-        let duration_fill = timeend_fill
-            .duration_since(timestart_fill)
-            .unwrap()
-            .as_secs();
-        println!("duration deserialize: {}", duration);
-        assert!(duration < 8);
-        println!("duration fill_vec: {}", duration_fill);
-        assert!(duration_fill < 8);
+        let duration_deserialize = time_end.duration_since(time_start).unwrap();
```

Comment:
I've got big reservations for the assertion part of this test. Why does it only test times? Why don't you check that the produced vectors take in the expected values, or at least in the expected count (plus maybe checking that each item is different)??

Reply:
It was ment to be performance test. If you insist, I can add more assertions.

Reply:
Yes, I thought it would be a nice enhancement as you haven't done it in any of those tests too. 

Also, if you want to put the stress on testing the performance it should somehow be seen in the name of the test. The name is quite general.